go/timer/timer.go

```diff
@@ -87,13 +87,23 @@ func (tm *Timer) Start(keephouse func()) {
 }
 
 func (tm *Timer) run(keephouse func()) {
+	var timer *time.Timer
+	stopTimer := func() {
+		if timer != nil {
+			timer.Stop()
+		}
+		timer = nil
+	}
+	defer stopTimer()
 	for {
+		stopTimer()
 		var ch <-chan time.Time
 		interval := tm.interval.Get()
 		if interval <= 0 {
 			ch = nil
 		} else {
-			ch = time.After(interval)
+			timer = time.NewTimer(interval)
+			ch = timer.C
 		}
 		select {
 		case action := <-tm.msg:
```

Comment:
How about stopping the timer here? Advantage is that timer can become a 0-inited value inside the `for` loop. So, it'll just be:
```
if timer != nil {
  timer.Stop()
}
```

Reply:
This is actually great idea which I looked for :)